### PR TITLE
APS File Upload

### PR DESCRIPTION
**Pull request by azaleacolburn**
### Description
Functions to interface with APS data management, the main purpose is to upload mirabuf files to a users's hub, so they can be easily accessed by fission.
Every user has a hub, in their hub they have a project, in which there are folders, in which there are buckets. Files are inside buckets.

### Objectives
- [x] Getters for hub ids, project ids, folder ids, file ids
- [x] Upload files to specific folder
- [x] Interface with the APS versioning system to upload new versions of mirabuf files

### Testing Done
- [x] Test with Hunter's janky solution (In Progress)
- [x] Test with #1001

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1721)

### CODE REVIEW

**Review by HunterBarclay — APPROVED** [review]
Works great. Slowly discovering more issues with the Simulator, now that we can bring in more bots.

**Comment by HunterBarclay** [comment]
@Ewie21 Make sure you format

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
~~There are some changes in `ConfigCommand.py` that seem to be friction exporting changes. Did some of the friction re-implementation get accidently pushed to this branch?~~

**Comment by azaleacolburn** [comment]
> There are some changes in `ConfigCommand.py` that seem to be friction exporting changes. Did some of the friction re-implementation get accidently pushed to this branch?

I removed a block of code that had already been commented out. The boolean changes were probably me testing to see what properties did. They shouldn't matter at all because all of that will get overriden by the friction override branch once it gets merged in though. I'll revert these changes just to be safe though.

File: exporter/SynthesisFusionAddin/src/UI/ConfigCommand.py [hunk 1]
```diff
@@ -1008,35 +1001,31 @@ def notify(self, args):
...
-                # save was canceled
-                return
+            if exporterOptions.exportLocation == ExportLocation.DOWNLOAD:
+                if isRobot:
+                    savepath = FileDialogConfig.SaveFileDialog(
+                        defaultPath=exporterOptions.fileLocation,
+                        ext="Synthesis File (*.synth)",
+                    )
+                else:
+                    savepath = FileDialogConfig.SaveFileDialog(defaultPath=exporterOptions.fileLocation)
```
Comment (BrandonPacewic) [inline exporter/SynthesisFusionAddin/src/UI/ConfigCommand.py]: - Note to my future self, there will be a conflict with #995.

File: exporter/SynthesisFusionAddin/src/UI/ConfigCommand.py [hunk 2]
```diff
@@ -1008,35 +1001,31 @@ def notify(self, args):
...
-                self.current.filePath = str(updatedPath)
+                if savepath == False:
+                    # save was canceled
+                    return
 
+                updatedPath = pathlib.Path(savepath).parent
+                if updatedPath != self.current.filePath:
+                    self.current.filePath = str(updatedPath)
+            else:
+                savepath = processedFileName
```
Comment (BrandonPacewic) [inline exporter/SynthesisFusionAddin/src/UI/ConfigCommand.py]: This `if` statement should be using the current selected export location not the one currently saved to the design. Which is what the `exporterOptions` object is here.

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]

**Review by BrandonPacewic — APPROVED** [review]
Still some problems with exporting a robot without APS and the docker container running. I believe that most of these problems were introduced in a different PR so no need to try and fix them here.

This is certainly good enough to get things started!

**Pull request merged**